src/compiler/checker.ts

```diff
@@ -8016,6 +8154,65 @@ module ts {
             }
         }
 
+        function getAwaitedType(type: Type, fallbackType?: Type, seen: boolean[] = []): Type {

```

Comment:
I have removed this and split `getAwaitedType` into two functions:

* `getPromisedType` - if the type is a promise, returns the **promised type** of the promise; otherwise, returns `undefined`.
* `getAwaitedType` - if the type is a promise, recursively unwraps the **promised type**; if it’s a thenable, reports an error; otherwise, returns the provided type.

This should make it clearer as to the intent for each usage.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Friday, February 13, 2015 5:08 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/checker.ts<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24709720>:

> @@ -8016,6 +8154,65 @@ module ts {

>              }

>          }

>

> +        function getAwaitedType(type: Type, fallbackType?: Type, seen: boolean[] = []): Type {

What is fallbackType supposed to be?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24709720>.